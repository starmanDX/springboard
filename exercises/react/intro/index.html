<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Part1</title>
</head>
<body>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="FirstComponent.jsx" type="text/jsx"></script>
    <script src="NamedComponent.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>