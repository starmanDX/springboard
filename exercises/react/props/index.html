<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>JSX Part2</title>
</head>
<body>
    <h1>JSX Props Part 2 - Tweets</h1>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="Tweet.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>