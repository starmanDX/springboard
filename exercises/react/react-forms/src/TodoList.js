import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid/dist";

function TodoList() {
  const INITIAL_STATE = [];
  const [todos, setTodos] = useState(INITIAL_STATE);
  const createTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
  };
  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };
  return (
    <div className="TodoList">
      <NewTodoForm createTodo={createTodo} />
      <ul>
        {todos.map((todo) => (
          <Todo
            key={todo.id}
            id={todo.id}
            removeTodo={removeTodo}
            task={todo.task}
          />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;
