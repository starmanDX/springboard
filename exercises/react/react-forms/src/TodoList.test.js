import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, task = "test the app") {
  const taskInput = todoList.getByLabelText("Add A Todo Task:");
  fireEvent.change(taskInput, { target: { value: task } });
  const button = todoList.getByText("Add New Todo Task");
  fireEvent.click(button);
}

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("creates a new todo task", function () {
  const todoList = render(<TodoList />);

  expect(todoList.queryByText("Remove Todo Task")).not.toBeInTheDocument();

  addTodo(todoList);
  const removeButton = todoList.getByText("Remove Todo Task");

  expect(removeButton).toBeInTheDocument();
  expect(removeButton.previousSibling).toContainHTML("<li>test the app</li>");

  expect(todoList.getAllByDisplayValue("")).toHaveLength(1);
});

it("removes a todo task", function () {
  const todoList = render(<TodoList />);

  addTodo(todoList);
  const removeButton = todoList.getByText("Remove Todo Task");

  fireEvent.click(removeButton);
  expect(removeButton).not.toBeInTheDocument();
});
