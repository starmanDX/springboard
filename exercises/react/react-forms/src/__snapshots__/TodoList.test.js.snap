// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <div
      class="NewTodoForm"
    >
      <form>
        <label
          for="task"
        >
          Add A Todo Task:
        </label>
        <input
          id="task"
          name="task"
          placeholder="Task"
          type="text"
          value=""
        />
        <button>
          Add New Todo Task
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
